PurchasesTests/Purchasing/OfferingsManagerTests.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  OfferingsManagerTests.swift
+//
+//  Created by Juanpe Catalán on 9/8/21.
+
+import XCTest
+import Nimble
+import StoreKit
+import PurchasesCoreSwift
+
+// TODO (post-migration): Move this test case to the new framework target (PurchasesCoreSwift).
+class OfferingsManagerTests: XCTestCase {
+
+    let mockDeviceCache = MockDeviceCache()
+    let mockOperationDispatcher = MockOperationDispatcher()
+    let mockSystemInfo = try! MockSystemInfo(platformFlavor: "iOS",
+                                             platformFlavorVersion: "3.2.1",
+                                             finishTransactions: true)
+    let mockBackend = MockBackend()
+    let mockOfferingsFactory = MockOfferingsFactory()
+    let mockProductsManager = MockProductsManager()
+    var offeringsManager: OfferingsManager!
+
+    override func setUp() {
+        super.setUp()
+        
+        offeringsManager = OfferingsManager(deviceCache: mockDeviceCache,
+                                            operationDispatcher: mockOperationDispatcher,
+                                            systemInfo: mockSystemInfo,
+                                            backend: mockBackend,
+                                            offeringsFactory: mockOfferingsFactory,
+                                            productsManager: mockProductsManager)
+    }
+
+}
+
+extension OfferingsManagerTests {
+
+    func testMissingProductDetailsReturnsNil() {
+        // given
```

Comment:
I've broken down writing the tests into three sections:

- The **given** section is the pre-conditions to the test.
- The **when** section is that behavior that you're specifying.
- The **then** section describes the changes you expect due to the specified behavior.

Please let me know if it looks fine for you :)

Reply:
I always try to follow given/when/then implicitly, I've never added explicit comments... but I like it. it makes the distinction pretty clear, particularly for tests that have more complicated setup. 
@taquitos thoughts?